actionmailer/lib/action_mailer/test_helper.rb

```diff
@@ -161,28 +162,17 @@ def assert_enqueued_email_with(mailer, method, params: nil, args: nil, queue: Ac
         mailer = mailer.instance_variable_get(:@mailer)
       end
 
-      expected_args = if args.is_a?(Proc) || params.is_a?(Proc)
-        -> (job_args) {
-          # Ordered from the least to the most heavy checks.
-          return false if [mailer.to_s, method.to_s, "deliver_now"] != job_args[0..2]
-          return false if !params.is_a?(Proc) && params != job_args[3][:params]
-          return false if !args.is_a?(Proc) && Array(args) != job_args[3][:args]
-          return false if args.is_a?(Proc) && !args.(job_args[3][:args])
-          return false if params.is_a?(Proc) && !params.(job_args[3][:params])
+      params, args = args, nil if args.is_a?(Hash)
+      args = Array(args) unless args.is_a?(Proc)
 
-          true
-        }
-      else
-        if args.is_a?(Hash)
-          [mailer.to_s, method.to_s, "deliver_now", params: args, args: []]
-        elsif params.present?
-          [mailer.to_s, method.to_s, "deliver_now", params: params, args: Array(args)]
-        else
-          [mailer.to_s, method.to_s, "deliver_now", args: Array(args)]
-        end
+      expected = ->(job_args) do
+        job_kwargs = job_args.extract_options!
```

Comment:
I believe `job_args` are passed in from the test body. Should we avoid modifying it in place, in case test author is trying to reuse it?

Reply:
`job_args` should be from Active Job rather than the test.  Specifically from `ActiveJob::Arguments.deserialize`:

https://github.com/rails/rails/blob/6adaeb864946718f71746958d3463e5c0a266b30/activejob/lib/active_job/test_helper.rb#L711

Which returns a new array via `map`:

https://github.com/rails/rails/blob/6adaeb864946718f71746958d3463e5c0a266b30/activejob/lib/active_job/arguments.rb#L43